lib/whitehall_importer/import.rb

```diff
@@ -84,7 +84,10 @@ def check_document_integrity(document)
     def check_edition_integrity(edition)
       integrity_checker = IntegrityChecker.new(edition)
 
-      raise AbortImportError, integrity_checker.problems.join(", ") unless integrity_checker.valid?
+      unless integrity_checker.valid?
+        state = edition.live? ? "live" : "draft"
+        raise WhitehallImporter::IntegrityCheckError.new(state, integrity_checker.problems, integrity_checker.proposed_payload)
```

Comment:
I think we can probably pass the integrity checker result into the `WhitehallImporter::IntegrityCheckError`. 

So something like: `raise WhitehallImporter::IntegrityCheckError.new, integrity_checker`

As it looks like we've got logic in the exception and before the exception.

Reply:
Now passing the state and `IntegrityChecker` instance.  It felt a bit weird to access the edition's state through the IntegrityChecker so keeping that as is.

Reply:
I don't think it's weird to do that since we have edition as a public method of IntegrityChecker - I think it's a bit weirder having the bits of logic on the calling side and in the initializer of the exception. We should probably also not use the word state as it means a different thing to what we normally mean by edition state.

Reply:
I've changed it to `publishing_status` - is that any better?